### DISCUSSION TITLE
Issue using findUnique in Typescript

### DISCUSSION THREAD

**Discussion by tomwray13** [prisma/prisma#8341] [Q&A] [answered]
Hey there!

I'm having an issue using the findUnique method with typescript.

I'm using a simple login resolver which takes an email as the only argument, and want to use the findUnique to find the respective user in the database.

Here's the code for the resolver:

```typescript
login: (parent: any, args: any) => {
    const findSpecificUser = (email: string) => {
        return Prisma.validator<Prisma.userWhereUniqueInput>()({
            email,
        })
    }
    const user = prisma.user.findUnique({ where: findSpecificUser(args.email) })
}
```

I'm getting a couple of errors show in VScode to do with the `userWhereUniqueInput` Prisma validator. It's saying: 

`Type '{ email: string; }' has no properties in common with type 'userWhereUniqueInput'.`

Hovering over the `userWhereUniqueInput` shows the following type:

```typescript
type Prisma.userWhereUniqueInput = {
    id?: number | undefined;
}
```

Which is confusing - it looks like the email is not included? Tried following [the docs](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#where) but can't figure it out!

**Comment by ryands17** [comment, marked as answer]
@tomwray13 👋 

You need to add `@unique` in front of the email field and then update your database via migrations like [this](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#unique-fields-as-unique-identifiers). This will make sure it appears in `userWhereUniqueInput`.

**Reply by tomwray13** [reply]
Doh! Thank you so much for the quick reply @ryands17!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
